File: packages/document-viewer/android/src/main/java/com/reactnativedocumentviewer/RNDocumentViewerPackage.java
// LICENSE: see License.md in the package root
package com.reactnativedocumentviewer;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.TurboReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.model.ReactModuleInfo;
import com.facebook.react.module.model.ReactModuleInfoProvider;

import java.util.HashMap;
import java.util.Map;

public class RNDocumentViewerPackage extends TurboReactPackage {

  @Nullable
  @Override
  public NativeModule getModule(String name, @NonNull ReactApplicationContext reactContext) {
    if (name.equals(RNDocumentViewerModule.NAME)) {
      return new RNDocumentViewerModule(reactContext);
    } else {
      return null;
    }
  }

  @Override
  public ReactModuleInfoProvider getReactModuleInfoProvider() {
    return () -> {
      boolean isTurboModule = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
      final Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
      moduleInfos.put(
        RNDocumentViewerModule.NAME,
        // deprecated in RN 0.73
        new ReactModuleInfo(
          RNDocumentViewerModule.NAME,
          RNDocumentViewerModule.NAME,
//          "DocumentViewerModule",
          false, // canOverrideExistingModule
          false, // needsEagerInit
          true, // hasConstants
          false, // isCxxModule
          isTurboModule // isTurboModule
        ));
      return moduleInfos;
    };
  }
}

File: packages/document-viewer/android/src/main/java/com/reactnativedocumentviewer/RNDocumentViewerPackage.kt
// LICENSE: see License.md in the package root
package com.reactnativedocumentviewer

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider

class RNDocumentViewerPackage : BaseReactPackage() {
  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == NativeDocumentViewerSpec.NAME) {
      RNDocumentViewerModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      val isTurboModule = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
      mapOf(
        NativeDocumentViewerSpec.NAME to ReactModuleInfo(
        NativeDocumentViewerSpec.NAME,
        NativeDocumentViewerSpec.NAME,  //          "DocumentViewerModule",
        false,  // canOverrideExistingModule
        false,  // needsEagerInit
        false,  // isCxxModule
        isTurboModule // isTurboModule
      ))
    }
  }
}
